lib/rubocop/cop/rails/squished_heredocs.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      #
+      # Checks SQL heredocs to use `.squish`
+      #
+      # @example
+      #
+      #   # bad
+      #   <<~SQL
+      #     SELECT * FROM table
+      #   SQL
+      #
+      #   # good
+      #   <<~SQL.squish
+      #     SELECT * FROM table
+      #   SQL
+      #
+      class SquishedHeredocs < Cop
+        include Heredoc
+        include RangeHelp
+
+        SQL = 'SQL'
+        SQUISH = '.squish'
+        SQL_SQUISH = "#{SQL}#{SQUISH}"
+        MSG = 'Use `%<expect>s` instead of `%<current>s`.'
+
+        def on_heredoc(node)
+          return if no_offense_detected?(node)
+
+          add_offense(node)
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.insert_after(heredoc_begin_line_range(node), SQUISH)
+          end
+        end
+
+        private
+
+        def no_offense_detected?(node)
+          no_sql_heredoc?(node) || using_squish_on_sql_heredoc?(node)
+        end
+
+        def no_sql_heredoc?(node)
+          delimiter_string(node) != SQL
+        end
```

Comment:
I'd invert this method for clarity, probably `no_offense_detected?` too.

```suggestion
        def sql_heredoc?(node)
          delimiter_string(node) == SQL
        end
```